akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -5,18 +5,25 @@
 package akka.http.scaladsl.model.ws
 
 import java.lang.Iterable
+
 import scala.collection.immutable
 import akka.NotUsed
+import akka.annotation.DoNotInherit
 import akka.stream._
 import akka.http.impl.util.JavaMapping
 import akka.http.javadsl.{ model ⇒ jm }
 import akka.http.scaladsl.model.HttpResponse
+import akka.japi.Pair
+import akka.stream.scaladsl.Flow
+
+import scala.concurrent.{ Future, Promise }
 
 /**
  * A custom header that will be added to an WebSocket upgrade HttpRequest that
  * enables a request handler to upgrade this connection to a WebSocket connection and
  * registers a WebSocket handler.
  */
+@DoNotInherit
```

Comment:
@raboof added the `@DoNotInherit` here and to the javadsl, as well as in the .excludes file, but still getting the binary compatibility error. Any ideas?

Reply:
That is just a documentation flag for users to not inherit, the bincomp checking tool does not know about it. You will need to add a filter entry to the right mima.excludes file (given that we are sure that this binary compatibility breakage is safe). You can see how to do that in the contributing guide: https://github.com/akka/akka-http/blob/master/CONTRIBUTING.md#binary-compatibility